runtime/lua/vim/ui/clipboard/osc52.lua

```diff
@@ -9,10 +9,34 @@ local function osc52(clipboard, contents)
   return string.format('\027]52;%s;%s\027\\', clipboard, contents)
 end
 
+---@class clipboard_cache
+---
+---@field lines_str string String representation of lines passed to the copy() funciton
+---@field regtype string regtype argument passed to the copy() function
+---
+---@type table<string, clipboard_cache>
+M.cache = {}
+
+-- When OSC 52 lines are the same as our cache, return our cache regtype
+local function get_regtype(reg, lines_str)
+  local cb_cache = M.cache[reg]
+  if cb_cache then
+    if cb_cache.lines_str == lines_str then
+      return cb_cache.regtype
+    end
+  end
+  return ''
+end
+
 function M.copy(reg)
   local clipboard = reg == '+' and 'c' or 'p'
-  return function(lines)
+  ---@param regtype string
+  return function(lines, regtype)
     local s = table.concat(lines, '\n')
+    M.cache[reg] = {
+      lines_str = s,
+      regtype = regtype,
+    }
```

Comment:
```suggestion
    cache[reg] = {
      hash = vim.fn.sha256(s),
      regtype = regtype,
    }
```

Reply:
Implemented in https://github.com/neovim/neovim/pull/34228/commits/1b89652e78afed3876bdbbc7a74bc16536929fec